CoreUtilities/src/au/gov/asd/tac/constellation/utilities/gui/field/framework/ConstellationInput.java

```diff
@@ -0,0 +1,645 @@
+     * ensure appropriate resizing with contained text.  
+     * The first layer is the background layer, a {@link Rectangle} which acts a a shape to provide the 
+     * fill color to an input field. When a field becomes invalid, it should be this 
+     * rectangle that changes color.
+     * The second Layer is the interactable content layer a {@link HBox} that hosts all of the "clickable"
+     * UI components. This layer need to have a clipping mask that this the same size as the foreground 
+     * to enable smooth corners. 
+     * The third layer is the foreground layer a {@link Rectangle} which acts as a boarder around the 
+     * input field to provide the focus effect for when the input is selected. 
+     */
+    private void buildInputFieldLayers(final ConstellationTextArea textArea){
+        final HBox interactableContentLayer = new HBox();
+        interactableContentLayer.setAlignment(Pos.TOP_CENTER);
+        HBox.setHgrow(interactableContentLayer, Priority.ALWAYS);
+
+        final Rectangle backgroundLayer = new Rectangle();
+        backgroundLayer.setArcWidth(corner);
+        backgroundLayer.setArcHeight(corner);  
+        backgroundLayer.setFill(Color.TRANSPARENT);
+        backgroundLayer.widthProperty().bind(interactableContentLayer.widthProperty());
+        
+        final Rectangle interactableContentClipingMask = new Rectangle();
+        interactableContentClipingMask.setArcWidth(corner);
+        interactableContentClipingMask.setArcHeight(corner);        
+        interactableContentClipingMask.widthProperty().bind(interactableContentLayer.widthProperty());
+        interactableContentLayer.setClip(interactableContentClipingMask);
+        
+        
+        final Rectangle foregroundLayer = new Rectangle();
+        foregroundLayer.setArcWidth(corner);
+        foregroundLayer.setArcHeight(corner);        
+        foregroundLayer.setFill(Color.TRANSPARENT);
+        foregroundLayer.setMouseTransparent(true);
+        foregroundLayer.widthProperty().bind(interactableContentLayer.widthProperty());
+        
+        textArea.bindHeightProperty(foregroundLayer, backgroundLayer, interactableContentClipingMask);
+        
+        this.getChildren().addAll(backgroundLayer, interactableContentLayer, foregroundLayer);
+    }
+    
+    /**
+     * To be called at the end of the constructor for extensions of this class.
+     * Builds out all of the implemented input features such as buttons, context, shortcuts and info windows.
+     * The framework of allowing each extension of the ConstellationInput to implement its own combination
+     * of input features leads to one slightly ugly requirement that this class cannot build its visual aspects
+     * until the extended class has finished initializing. If not done this way, we get some ugly errors where methods and 
+     * objects don't exist...  
+     */
+    protected void initialiseDepedantComponents(){
```

Comment:
if this function is running a series of `if (this instanceof ...)` would it make more sense to make this an abstract function and then have relevant subtypes override?